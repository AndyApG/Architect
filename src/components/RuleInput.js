/* eslint-disable jsx-a11y/label-has-for */

import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

const RuleInput = ({
  value,
  type,
  onChange,
  className,
}) => (
  <label className={cx('rule-input', className)}>
    <div className="rule-input__spacer">{ value }</div>
    <input
      className="rule-input__text"
      type={type}
      value={value}
      onChange={onChange}
    />
  </label>
);

RuleInput.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
  className: PropTypes.string,
  type: PropTypes.oneOf(['text', 'number']),
};

RuleInput.defaultProps = {
  className: null,
  onChange: () => {},
  value: '',
  type: 'text',
};

export default RuleInput;
