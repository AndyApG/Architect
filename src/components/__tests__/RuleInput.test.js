/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import RuleInput from '../RuleInput';

const mockProps = {
  value: '',
  onChange: () => {},
};

describe('<RuleInput />', () => {
  it('can render', () => {
    const component = shallow(<RuleInput {...mockProps} />);

    expect(component).toMatchSnapshot();
  });

  it('can render as number', () => {
    const options = {
      ...mockProps,
      type: 'number',
    };

    const component = shallow(<RuleInput {...options} />);

    expect(component).toMatchSnapshot();
    expect(component.find('input').prop('type')).toEqual('number');
  });

  it('can prepopulate a value', () => {
    const options = {
      ...mockProps,
      value: 'foo',
    };

    const component = shallow(<RuleInput {...options} />);

    expect(component).toMatchSnapshot();

    expect(component.find('input').prop('value')).toEqual('foo');
  });

  it('it triggers onChange with value', () => {
    const onChange = jest.fn();
    const options = {
      ...mockProps,
      onChange,
    };

    const component = shallow(<RuleInput {...options} />);

    component.find('input').simulate('change', { target: { value: 'bar' } });

    expect(onChange.mock.calls).toEqual([['bar']]);
  });
});
