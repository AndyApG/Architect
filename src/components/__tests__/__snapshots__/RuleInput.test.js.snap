// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RuleInput /> can prepopulate a value 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RuleInput
    className={null}
    onChange={[Function]}
    type="text"
    value="foo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="rule-input__spacer"
>
          foo
</div>,
        <input
          className="rule-input__text"
          onChange={[Function]}
          type="text"
          value="foo"
/>,
      ],
      "className": "rule-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "foo",
          "className": "rule-input__spacer",
        },
        "ref": null,
        "rendered": "foo",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "rule-input__text",
          "onChange": [Function],
          "type": "text",
          "value": "foo",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="rule-input__spacer"
>
            foo
</div>,
          <input
            className="rule-input__text"
            onChange={[Function]}
            type="text"
            value="foo"
/>,
        ],
        "className": "rule-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "foo",
            "className": "rule-input__spacer",
          },
          "ref": null,
          "rendered": "foo",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "rule-input__text",
            "onChange": [Function],
            "type": "text",
            "value": "foo",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<RuleInput /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RuleInput
    className={null}
    onChange={[Function]}
    type="text"
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="rule-input__spacer"
/>,
        <input
          className="rule-input__text"
          onChange={[Function]}
          type="text"
          value=""
/>,
      ],
      "className": "rule-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "rule-input__spacer",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "rule-input__text",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="rule-input__spacer"
/>,
          <input
            className="rule-input__text"
            onChange={[Function]}
            type="text"
            value=""
/>,
        ],
        "className": "rule-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "rule-input__spacer",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "rule-input__text",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<RuleInput /> can render as number 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RuleInput
    className={null}
    onChange={[Function]}
    type="number"
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="rule-input__spacer"
/>,
        <input
          className="rule-input__text"
          onChange={[Function]}
          type="number"
          value=""
/>,
      ],
      "className": "rule-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "rule-input__spacer",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "rule-input__text",
          "onChange": [Function],
          "type": "number",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="rule-input__spacer"
/>,
          <input
            className="rule-input__text"
            onChange={[Function]}
            type="number"
            value=""
/>,
        ],
        "className": "rule-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "rule-input__spacer",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "rule-input__text",
            "onChange": [Function],
            "type": "number",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
